opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -254,12 +257,13 @@ def _check_attribute_value_sequence(sequence: Sequence) -> Optional[str]:
 
         first_element_type = type(sequence[0])
 
-        if issubclass(first_element_type, Number):
-            first_element_type = Number
-
-        if first_element_type not in (bool, str, Number):
+        if first_element_type not in (bool, str, int, float):
             return "invalid type"
 
+        # int and float are both numeric types, allow mixed arrays of those
```

Comment:
This and `Sequence[Union[int, float]]` above: do we really want to allow mixed-type array attributes?

Reply:
Good point. https://github.com/open-telemetry/opentelemetry-python/pull/348 added a specific test for that. The specification is not that clear, it talks about `numeric` as a primitive type [1]. Having a second thought, float and int are treated different in the exporters, so I think it would make sense not to avoid mixed arrays of those. If you agree I could update this.

[1] https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#set-attributes.

Reply:
I don't want this to block the PR, up to you whether to update this in this PR or another one.

Reply:
I updated the PR to avoid that kind of mixed attributes.